src/Shared/Recruit.Vacancies.Client/Infrastructure/Mongo/MongoDbCollectionBase.cs

```diff
@@ -28,5 +38,33 @@ protected IMongoCollection<T> GetCollection<T>()
 
             return collection;
         }
+
+        protected Task Retry(Func<Task> func)
+        {
+            var policy = GetRetryPolicy();
+
+            return policy.ExecuteAsync(func);
+        }
+
+        protected Task<T> Retry<T>(Func<Task<T>> func)
+        {
+            var policy = GetRetryPolicy();
+
+            return policy.ExecuteAsync(func);
+        }
+
+        private Polly.Retry.RetryPolicy GetRetryPolicy()
```

Comment:
Should we inject the policy in?  Having a back-off policy is OK for Jobs but for the web we might want to fail quicker.  Maybe we should revisit this after we've released.

Reply:
Yes I agree we should. Have added an item on tech debt list.